src/App.jsx

```diff
@@ -1,248 +1,337 @@
+
+  if (filterProducts) {
+    if (filterProducts !== USERS.ALL) {
+      copiedProduct = copiedProduct.filter(
+        copieProduct => copieProduct.user.name === filterProducts,
+      );
+    }
+  }
+
+  if (qwery) {
+    const normalizedQwery = qwery.toLowerCase();
+
+    copiedProduct = copiedProduct.filter(
+      copieProduct => copieProduct.name.toLowerCase().includes(normalizedQwery),
+    );
+  }
+
+  if (filterCategories) {
+    if (filterCategories !== CATEGORY.ALL) {
+      copiedProduct = copiedProduct.filter(
+        copieProduct => copieProduct.category.title === filterCategories,
+      );
+    }
+  }
+
+  return copiedProduct;
+}
+
+export const App = () => {
+  const [filterProducts, setfilterProducts] = useState(USERS.ALL);
+  const [qwery, setQwery] = useState('');
+  const [filterCategories, setFilterCategories] = useState(CATEGORY.ALL);
+
+  const handleReset = () => {
+    setfilterProducts('');
+    setQwery('');
+    setFilterCategories('');
+  };
+
+  const filteredProducts = getFIlteredProducts(
+    products, { filterProducts, qwery, filterCategories },
+  );
+
+  return (
+    <div className="section">
+      <div className="container">
+        <h1 className="title">Product Categories</h1>
+        <div className="block">
+          <nav className="panel">
+            <p className="panel-heading">Filters</p>
+
+            <p className="panel-tabs has-text-weight-bold">
+              <a
+                data-cy="FilterAllUsers"
+                href="#/"
+                onClick={() => setfilterProducts(USERS.ALL)}
+                className={`${filterProducts === USERS.ALL ? 'is-active' : ''}`}
+              >
+                All
+              </a>
+
+              <a
+                data-cy="FilterAllUsers"
+                href="#/"
+                onClick={() => setfilterProducts(USERS.ROMA)}
+                className={`${filterProducts === USERS.ROMA ? 'is-active' : ''}`}
+              >
+                Roma
+              </a>
 
-                  <a href="#/">
-                    <span className="icon">
-                      <i data-cy="SortIcon" className="fas fa-sort" />
-                    </span>
-                  </a>
-                </span>
-              </th>
-            </tr>
-          </thead>
-
-          <tbody>
-            <tr data-cy="Product">
-              <td className="has-text-weight-bold" data-cy="ProductId">
-                1
-              </td>
-
-              <td data-cy="ProductName">Milk</td>
-              <td data-cy="ProductCategory">🍺 - Drinks</td>
-
-              <td
-                data-cy="ProductUser"
-                className="has-text-link"
+              <a
+                data-cy="FilterUser"
+                href="#/"
+                onClick={() => setfilterProducts(USERS.ANNA)}
+                className={`${filterProducts === USERS.ANNA ? 'is-active' : ''}`}
+              >
+                Anna
+              </a>
```

Comment:
1. These elements are the same, you could create a list with user names and render these elements using `rending lists`
2. Also seems like incorrect `data-cy` attribute here
```
<a
  data-cy="FilterAllUsers"
  href="#/"
  onClick={() => setfilterProducts(USERS.ROMA)}
  className={`${filterProducts === USERS.ROMA ? 'is-active' : ''}`}
>
  Roma
</a>
```
3. 🚨 The same with categories